nix/lib/github/mkPushResult.nix

```diff
@@ -0,0 +1,45 @@
+utils: lib: {
+  mkPushResult = {
+    owner,
+    repo,
+    branch,
+    patches ? [],
+  }:
+    lib.builders.mkActionScript {
+      mkPath = system: let
+        pkgs = utils.pkgs.${system};
+      in [pkgs.jq pkgs.git];
+      mkScript = system: ''
+        stdin=$(cat)
+
+        input=$(echo "$stdin" | jq -r '.input | ${utils.jqJsonToBashArray}')
+        declare -A input="($input)"
+
+        [ "''${input[status]}" == "success" ] || { echo "Unsuccessful build: nothing to push" >&2; exit 0; }
+
+        token=$(echo "$stdin" | jq -r '.secrets.github_token')
+
+        cp -r "''${input[out]}" out
+        chmod -R +w out
+        cd out
+
+        git init
+        git config user.email "typhon@typhon-ci.org"
+        git config user.name "Typhon"
+        export GIT_AUTHOR_DATE="1970-01-01T00:00:00+0000"
+        export GIT_COMMITTER_DATE="1970-01-01T00:00:00+0000"
```

Comment:
Would be nice to put that in utils, what do you think?

Reply:
actually it would be even nicer to have `lib.builders.mkActionScript` set this up, in the spirit of #38

Reply:
this is beyond the scope of this PR anyway, so for the time being I moved this to `lib.git.config`